### DISCUSSION TITLE
Series and Characters

### DISCUSSION THREAD

**Discussion by paulkater** [vkbo/novelWriter#1946] [Q&A]
I tend to fall into writing series a lot. This means recurring characters.

One of my series has 26 volumes so far, which says something about numbers, I guess.

I wonder if I would break anything in novelWriter if I were to set up a character repository (simply a directory) with .md files, where I keep the character descriptions, and then (oh boy)...
- create an empty character file in a new volume
- close novelWriter
- figure out the name of the file on disk
- and replace that with a symlink (I use Linux) to the character file in the repo

That would get me the character details in every new volume I start, and each project would have the latest info, also the older ones. (I keep track of which detail I added in which book / chapter, and sometimes even which scene.)

I've noticed hashes and other meta-data in the original files, and I don't want to crash anything, hence me asking before going for it and damaging anything.

It would be awesome if this works; it's something I haven't found in any other writing software yet (and I've tried lots of them).

Maybe others, who also write series, have a better way of doing this. If so, I'm all eyes (instead of ears, as I'll be reading this).

**Comment by vkbo** [comment]
TLDR: Yes, what you outline should work just fine, but you have to trigger an index update manually if the content changes.

In more detail:

As of 2.0, novelWriter does support multi-novel projects. Personally, my stories tend towards trilogies because I like to get the most out of my world building. You can create new "Novel" folders for each volume. The "Novel View" and "Outline" panels will then allow you to select which novel it displays the meta data for.

That said, I see that it may be an issue for as many as 26 novels in a single project. I have never stress tested novelWriter at that scale. The main thing it struggles with is very large single documents, but that has improved a fair bit in the latest releases. My stress test document has also generally been 800k words, which is more than pretty much anything encountered in real use cases, and it handles that just fine these days. I'm not sure how it would handle 26 novels of scene files. I should actually generate a test case for that.

In any event, the meta data at the top of the file is mainly there to assist in recovery. You can safely delete it at any time and nothing will break as long as the file is already a member of the project. On the flip side of this, you can also copy documents between projects and they will be added to the project on startup based on this meta data. So you don't even need to create your empty char file. You can simply symlink it with the existing file name key. It should automatically import the file in the first available Character folder.

The meta data hash is only used to detect cases where you edit a document externally while it is being edited in the app. That's the only time the hash is checked. It is only there to ensure it captures delayed file syncs, and all it does is issue an overwrite warning if it is about to overwrite text that is different than what it read on document open. It's just a warning, and you can click "Yes" to continue.

The only issue you may run into with external edit of a given project is that the index is not automatically updated until you open and save the document. You can regenerate the whole index though by pressing F9 or from the Tools menu.

**Reply by paulkater** [reply]
That sounds fabulous, thank you for this detailed answer. Sounds like perfect to me. :smiley: 
I'm not sure if I will ever manifest another 26 volume series (this one started 14 years ago), but I'm happy with your reply and I'm going to play with this.
Thank you again.
Paul
